src/DurableTask.Netherite/StorageLayer/Faster/FasterKV.cs

```diff
@@ -2076,7 +2081,15 @@ bool IFunctions<Key, Value, EffectTracker, Output, object>.SingleWriter(ref Key
                         break;
 
                     case WriteReason.CopyToTail:
-                        this.cacheDebugger?.Record(key.Val, CacheDebugger.CacheEvent.SingleWriterCopyToTail, src.Version, default, info.Address);
+                        takeValueFromOutput = (output.Val != null); // we have observed that src.Val is null sometimes, so if present, we use output instead
+                        if (takeValueFromOutput)
+                        {
+                            this.cacheDebugger?.Record(key.Val, CacheDebugger.CacheEvent.SingleWriterCopyToTailFromOutput, output.Version, default, info.Address);
+                        }
+                        else
+                        {
+                            this.cacheDebugger?.Record(key.Val, CacheDebugger.CacheEvent.SingleWriterCopyToTail, src.Version, default, info.Address);
+                        }
```

Comment:
I think this is a good place to elaborate a bit more on the significance of `src.Val` being null. Is this a bug in FASTER, or expected, or something else? Can you please add a small blurb about that here?

Reply:
That's sort of what the comment on line 2084 is supposed to be.

Reply:
I'm not sure that gives me enough context though. The way the comment is phrased ("we have observed...sometimes") makes me thing it is not expected that `src.Val` would be null, and if that's true then I think that's worth calling out explicitly. It would tell me whether or not there's something weird that FASTER is doing here